src/config/mod.rs

```diff
@@ -466,10 +470,17 @@ pub struct TransformOuter {
     pub inner: Box<dyn TransformConfig>,
 }
 
+pub type EnrichmentTableList =
+    Arc<RwLock<HashMap<String, Box<dyn enrichment_tables::EnrichmentTable + Send + Sync>>>>;
```

Comment:
I would like to have used `tokio::sync::RwLock` here, but the `transform` method of `FunctionTransform` isn't `async` so I would be unable to read from the lock in that function. 

This does mean that the whole thread will block whilst the the source is locked for writing. For now this is unlikely to be an issue as reloading the data will be a pretty rare occurrence, but we may want to look into this in due course.

Reply:
Thinking about it, the whole of VRL isn't async anyway, so using tokio here would be much larger change anyway..

Reply:
Something like `evmap` could be a good fit here since it should be read-dominated. Also a little simpler.

Reply:
> Something like `evmap` could be a good fit here

I don't think `evmap` is going to work here. The main issue is that [`FunctionTransform`](https://github.com/timberio/vector/blob/master/lib/vector-core/src/transform/mod.rs#L104) needs to be `Send + Sync`.

[`evmap::ReadHandle`](https://docs.rs/evmap/10.0.2/evmap/struct.ReadHandle.html) is unfortunately not `Sync`. So the remap transform would need to store a [`evmap::ReadHandleFactory`](https://docs.rs/evmap/10.0.2/evmap/struct.ReadHandleFactory.html) which is `Sync`. remap could get a `ReadHandle` from this when it needs to. 

The problem is that to get a `ReadHandle` from the factory requires taking a lock. As per the docs:

>  Note that this internally takes a lock whenever you call ReadHandleFactory::handle, so you should not expect producing new handles rapidly to scale well.

This unfortunately obliterates any advantage that we may have from using `evmap`.